project/MiMaFilters.scala

```diff
@@ -18,6 +18,7 @@ object MiMaFilters {
     exclude[MissingClassProblem]("scala.annotation.experimental"),
     exclude[MissingClassProblem]("scala.annotation.internal.ErasedParam"),
     exclude[MissingClassProblem]("scala.annotation.internal.ErasedParam"),
+    exclude[MissingClassProblem]("scala.annotation.internal.ProvisionalSuperClass"),
```

Comment:
Btw there's glob support in mima's exclusions so you can use `scala.annotation.internal.*` here next time.